### PR TITLE
Added documentation for other shared resources.

### PR DESCRIPTION
**Pull request by manthanand**
I want to use this as an example of what a documentation PR should look like. If you see something missing or are confused by the wording, then let me know. Also, leave comments if you think I should change the formatting of how I wrote the documentation.

### CODE REVIEW

File: Docs/source/Software/Apps.rst [hunk 1]
```diff
@@ -260,10 +260,66 @@ MinionsIO Semaphore
...
+waiting for this :term:`semaphore <Semaphore>`. Whenever the SPI transfer is complete, we must signal this semaphore so we need to 
 have an :term:`ISR <ISR>` that calls the signal semaphore function whenever a transfer is complete. This ISR is in the BSP SPI module.
 
+Safety Check Semaphore
+======================
+
+When the :ref:`Critical State <Critical state Task: Manthan Upadhyaya>` runs, it checks to see if the Amperes, Voltage, Open Wire,
+and Temperature are safe. This process is done by the Critical State Task using a blocking Pend function on the Safety Check Semaphore.
+Each time a semaphore is posted, its ``cnt`` parameter is incremented. When it is pended, the ``cnt`` parameter is decremented. The task
+uses a blocking pend (wont move past the function until the semaphore is pended ``NUM_FAULT_POINTS`` amount of times).
```
Comment (ClarkPoon) [inline Docs/source/Software/Apps.rst:272]: I don't think it makes sense to put a general explanation of how semaphores work here. If we want to include this in the docs, maybe we should have a section for explaining general concepts? If not we can link to some external resources like the uCOS-III docs

File: Docs/source/Software/Apps.rst [hunk 2]
```diff
@@ -260,10 +260,66 @@ MinionsIO Semaphore
...
+
+Used in `CAN_Queue.c` to flag that there is data in the :term:`CAN` :term:`FIFO`. Posted by tasks that place data 
+in the CAN FIFO and pended when data is removed from the FIFO.
+
+CAN Mailbox Semaphore
+=====================
+
+There are 3 hardware mailboxes that can hold CAN messages that will be sent along the CAN line. This semaphore is 
+initialized to a value of 3 (number of empty mailboxes). When the semaphore is posted, the count decreases (one 
+mailbox becomes full). When the semaphore is pended, the count increases (one mailbox becomes empty).
```
Comment (ClarkPoon) [inline Docs/source/Software/Apps.rst:298]:
```suggestion
There are 3 hardware mailboxes that can hold CAN messages that will be sent along the CAN line. This semaphore is 
initialized to a value of 3 (number of empty mailboxes). When the semaphore is posted, the count decreases (one 
mailbox becomes full). When the semaphore is pended, the count increases (one mailbox becomes empty). This means that the semaphore will keep track of how many CAN mailboxes are free. If we pend this semaphore before sending a CAN message and post it after sending a CAN message, we can make sure the code will block until a mailbox is free if we try to send a CAN message when no mailboxes are free.
```

File: Docs/source/Software/Apps.rst [hunk 3]
```diff
@@ -260,10 +260,66 @@ MinionsIO Semaphore
...
+
 Voltage Mutex
 =============
 
 Mutually excludes accesses to the Voltage buffer in the Voltage.c library. 
+
+MinionsASIC_Mutex
+=================
+
+This mutex excluses access to the ``Minions`` struct. This struct hold voltage and temperature data that is taken from the LTC6811.
```
Comment (ClarkPoon) [inline Docs/source/Software/Apps.rst:325]: `Minions` is an array of structs. Also, this sentence has typos

File: Docs/source/Software/Apps.rst (hunk 1 above)
Comment (manthanand) [reply Docs/source/Software/Apps.rst:272]: Yea, this is included in the glossary as well.

**Review by tiandahuang — APPROVED** [review]
looks good :)

**Pull request merged**